generator/generic/generic_generator.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative '../generator_base'
+require_relative '../sequence_metadata'
+require_relative './read_test'
+require_relative './profile_validation_test'
+
+module Inferno
+  module Generator
+    class GenericGenerator < Generator::Base
+      include ReadTest
+      include ProfileValidationTest
+
+      def resource_profiles
+        resources_by_type['StructureDefinition'].reject { |definition| definition['type'] == 'Extension' }
+      end
+
+      def sequence_metadata
+        @sequence_metadata ||= resource_profiles.map { |profile| SequenceMetadata.new(profile) }
+      end
+
+      def generate
+        generate_sequences
+        generate_module
+      end
+
+      def generate_sequences
+        sequence_metadata.each do |metadata|
+          create_read_test(metadata)
+          create_profile_validation_test(metadata)
+          generate_sequence(metadata)
+        end
+      end
+
+      def generate_sequence(metadata)
+        puts "Generating #{metadata.title}\n"
+        file_name = File.join(sequence_out_path, metadata.file_name + '.rb')
+        template = ERB.new(File.read(File.join(__dir__, 'templates/sequence.rb.erb')))
+        output =   template.result_with_hash(metadata: metadata)
+        FileUtils.mkdir_p(sequence_out_path + '/') unless File.directory?(sequence_out_path + '/')
+        File.write(file_name, output)
+      end
+
+      def generate_module
+        file_name = "#{module_yml_out_path}/#{@path}_module.yml"
```

Comment:
It would be helpful for me if this were its own method.

Reply:
The module_info part?

Reply:
The filename, so that we can easily locate the generated module file elsewhere.